### PR TITLE
[Vectorized][Function] Support all geolocation functions on vectorized engine

### LINKED ISSUE #8788: [Feature] Support geolocation functions on vectorized engine
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, geolocation functions are not supported on vectorized engine.

I will implement geolocation functions and split those to two PRs which is more easier for reviewing. 

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #8788 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/functions_geo.cpp [hunk 1]
```diff
@@ -199,13 +206,338 @@ struct StDistanceSphere {
...
+                    (Impl::shape_type != GEO_SHAPE_ANY && shape->type() != Impl::shape_type)) {
+                    res->insert_data(nullptr, 0);
+                    continue;
+                }
+                buf.clear();
+                shape->encode_to(&buf);
+                res->insert_data(buf.data(), buf.size());
+            }
+            block.replace_by_position(result, std::move(res));
+            return Status::OK();
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.cpp:472]: here to many `return Status::OK();` should be only one out of `if`

File: be/src/vec/functions/functions_geo.cpp [hunk 2]
```diff
@@ -199,13 +206,338 @@ struct StDistanceSphere {
...
+        const auto size = geo->size();
+        auto null_type = std::reinterpret_pointer_cast<const DataTypeNullable>(return_type);
+        auto res = ColumnNullable::create(return_type->create_column(), ColumnUInt8::create());
+
+        StConstructState* state = (StConstructState*) context->get_function_state(FunctionContext::FRAGMENT_LOCAL);
+        if (state == nullptr) {
+            GeoParseStatus status;
+            std::string buf;
+            for (int row = 0; row < size; ++row) {
+                if (geo->is_null_at(row)) {
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.cpp:455]: why need check null in here ? Null Frameworks should help to do this ？

File: be/src/vec/functions/functions_geo.cpp [hunk 3]
```diff
@@ -199,13 +206,338 @@ struct StDistanceSphere {
...
+            res->insert_data(nullptr, 0);
+            block.replace_by_position(result, ColumnConst::create(std::move(res), size));
+            return Status::OK();
+        }
+
+        StContainsState local_state;
+        int i;
+        GeoShape* shapes[2] = {nullptr, nullptr};
+        for (int row = 0; row < size; ++row) {
+            if (shape1->is_null_at(row) || shape2->is_null_at(row)) {
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.cpp:332]: why need check null in here ? Null Frameworks should help to do this ？

File: be/src/vec/functions/functions_geo.cpp [hunk 4]
```diff
@@ -199,13 +206,338 @@ struct StDistanceSphere {
...
+                if (value != GEO_PARSE_OK) {
+                    res->insert_data(nullptr, 0);
+                    continue;
+                }
+                buf.clear();
+                circle.encode_to(&buf);
+                res->insert_data(buf.data(), buf.size());
+            }
+            block.replace_by_position(result, std::move(res));
+            return Status::OK();
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.cpp:250]: too many return `OK`

File: be/src/vec/functions/functions_geo.cpp [hunk 5]
```diff
@@ -199,13 +206,336 @@ struct StDistanceSphere {
...
+
+struct StContains {
+    static constexpr auto NEED_CONTEXT = true;
+    static constexpr auto NAME = "st_contains";
+    static const size_t NUM_ARGS = 2;
+    static Status execute(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                          size_t result) {
+        DCHECK_EQ(arguments.size(), 2);
+        auto return_type = block.get_data_type(result);
+        if (return_type->is_nullable()) {
```
Comment (HappenLee) [inline be/src/vec/functions/functions_geo.cpp:316]: use `remove_nullable` to replace the code

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**